authors/apps/articles/views.py

```diff
@@ -313,4 +325,32 @@ def get(self, request):
             return Response({
                 "reports": reported}, status.HTTP_200_OK)
         return Response(
-            {"error": "You do not have permission to view the reported articles"}, status.HTTP_403_FORBIDDEN)
+            {"error": "You do not have permission to view the reported articles"},
+            status.HTTP_403_FORBIDDEN)
+
+
+class AddRead(generics.CreateAPIView):
+    permission_classes = (IsAuthenticated,)
+    serializer_class = UserReadsSerializer
+
+    def create(self, request, **kwargs):
+        user = request.user
+        article_id = request.data.get('article_id')
+        if not article_id:
+            return Response({'error': 'You must provide an article id to proceed'},
+                            status.HTTP_400_BAD_REQUEST)
+        try:
+            read = UserReads.objects.get(article=article_id, user=user.id)
+            if read:
```

Comment:
Why are we limiting a read on only one read? Imagine a scenario where we want to recommend articles to a particular user given their reading behavior. One of the best possible ways to make accurate recommendations is by looking at the stories they read more than once in relation to their tags.

Reply:
I added the restriction so that we can have a fairly accurate result when fetching the views. This is to ensure that scenerios such as if a user plots to view an article countless times for purposes of increasing their views we do not count those.